### ISSUE TITLE
Having trouble trying to `git pull`

### ISSUE THREAD

**Issue by ashleyy2444** [nus-cs2103-AY2324S2/forum#829]
Previously, I was facing this issue: 
https://stackoverflow.com/questions/62385234/git-keeps-duplicating-files-and-appending-2-at-end-of-filename#:~:text=If%20you%20have%20your%20git,with%20your%20git%20version%20control.

Now, I have deleted all my duplicate files successfully, but I am still unable to `git pull` to get the latest version. It kept giving this error message as shown:

![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/122223556/07cc20e9-c330-4d0c-9d9a-ad1028bdbea6)

When searching for the branch, there was 'branch-addConditionForDelete', but not 'branch-addConditionForDelete 2'. Since my PR was merged already, I decided to delete this branch 'branch-addConditionForDelete'. However, the problem still persists. 

I tried to clone again, but I was still unable to `git pull` the latest version. 

May I know how to resolve this issue?

**Comment by lhy-hoyin** [comment]
@ashleyy2444 When you say

> When searching for the branch, there was 'branch-addConditionForDelete', but not 'branch-addConditionForDelete 2'.

are you referring to ... ?
- your team's repo on GitHub
- your fork of the team repo (on GitHub also)
- your local repo (offline; on IntelliJ)

**Comment by lhy-hoyin** [comment]
Since your PR has been merged already and you wish to delete the branch `branch-addConditionForDelete` (and by extension, `branch-addConditionForDelete 2` too), you may want to make sure that you have actually delete all of such branches across all repos (i.e, team, fork and local).

[deleting branches within your repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository#deleting-a-branch)

**Comment by francisyzy** [comment]
Have you tried making a new folder and doing `git clone`?

You can also look into deleting branches on GitHub.com on this page https://github.com/ashleyy2444/tp/branches/all

**Comment by seanfirefox** [comment]
Can you try to `git branch` to check the local names of your github branches? Then verify if all of your branches are indeed appended with `  2` at the back. If it is, rename your branches using `git branch -m new-branch-name` while you are on the correct branch. Attempt to `git pull` again.

**Comment by E0735389** [comment]
The issue is this.

https://stackoverflow.com/questions/72515916/git-fatal-bad-object-refs-heads-2-master

Your `.git/` directory is corrupted. Best course of action at the moment appears to be to delete the repository and `git clone` a new one. Then disable iCloud in that directory.

Or you can try to delete the offending files manually (I think it's not worth it though)

**Comment by damithc** [comment]
> The issue is this.
> 
> https://stackoverflow.com/questions/72515916/git-fatal-bad-object-refs-heads-2-master
> 
> Your `.git/` directory is corrupted. Best course of action at the moment appears to be to delete the repository and `git clone` a new one. Then disable iCloud in that directory.
> 
> Or you can try to delete the offending files manually (I think it's not worth it though)

Good find, @E0735389 

In general, it is not a good idea to put a Git repo in a folder that is also auto-synced by other tools such as dropbox, icloud, onedrive etc.

**Comment by jinhanfromNUS** [comment]
Might not be very relevant, but I think you can force reset your local machine master branch to local repo master branch by running below at command prompt.

To force a git pull of the master branch to your local master branch, you can use the following command:
git fetch origin master     # Fetches the latest changes from the remote master branch
git reset --hard origin/master  # Resets your local master branch to match the remote master branch

**Comment by E0735389** [comment]
> Might not be very relevant, but I think you can force reset your local machine master branch to local repo master branch by running below at command prompt.
> 
> To force a git pull of the master branch to your local master branch, you can use the following command: git fetch origin master # Fetches the latest changes from the remote master branch git reset --hard origin/master # Resets your local master branch to match the remote master branch

No, it wouldn't work once the `.git` directory is corrupted. You can try it yourself.

```
$ cp ./.git/refs/heads/master ./.git/refs/heads/"master 2"  # intentionally corrupt the folder structure
$ git fetch origin master
fatal: bad object refs/heads/master 2
error: https://github.com/user/repo did not send all necessary objects
```
